poncho/src/poncho_package_run

```diff
@@ -152,6 +152,17 @@ function cleanup {
     s=$((total_ms/1000))
 
     logmsg total runtime: $(printf "%d.%0.3d seconds" "${s}" "${ms}")
+    cpids=$(pgrep -P "$$") # use pgrep to get a list of all associated child processes
+    for cpid in $cpids; # loop through and send termination signals to all children
+    do
+        kill -SIGTERM $cpid
+    done
+    # WARNING!!!!! this is unfinished. Any child process that blocks/ignores SIGTERM will cause this program to wait infinitely for them to finish
+    # the proper solution would be to set a timer that sends an interrupt signal to this process after a certain wait (say 60 seconds)
+    # after that time, if any child processes remain running, send SIGKILL to them because that can not be blocked and attempt to wait again for them to return
+    # unfortunately, bash has not native implementation of alarm(), and the only way for a process to send a signal to itself in the future is to create another background process to handle that
+    # but creating another process to send a signal to oneself has its own host of issues...
+    wait $cpids
```

Comment:
One thing to consider is that there is no way to wait for grandchildren processes, so the use of pgrep together with wait may not work as we want. We can only wait on the child process that we spawn when running the command line.